File: server/src/responsible/repositories/ResponsibleRepository.ts
import { Responsible as ResponsiblePrisma } from '@prisma/client';

import { CreateResponsibleDTO } from '@/responsible/dtos/CreateResponsibleDTO';

export abstract class ResponsibleRepository {
  abstract create(data: CreateResponsibleDTO): Promise<ResponsiblePrisma>;
  abstract findById(id: string): Promise<ResponsiblePrisma | null>;
  abstract findByEmail(email: string): Promise<ResponsiblePrisma | null>;
}

File: microservices/responsible-service/src/main/java/com/sportsfinance/responsible/domain/repository/ResponsibleRepository.java
package com.sportsfinance.responsible.domain.repository;

import com.sportsfinance.responsible.api.dto.CreateResponsibleDTO;
import com.sportsfinance.responsible.domain.model.Responsible;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ResponsibleRepository  extends JpaRepository<Responsible, String> {

    Responsible findByEmail(String email);
    CreateResponsibleDTO findByResponsibleId(String responsibleId);

}
